### ISSUE TITLE
Icons dissappearing randomly

### ISSUE THREAD

**Issue by h0uter** [FlorianWoelki/obsidian-icon-folder#59]
**Describe the bug**
Icons randomly disappear and instead their shortcodes are displayed.

**To Reproduce**
1. it happens randomly

However, once its happened I can temporariliy fix it by deleting and reinstalling the icons packs.
However, this only works if all my folders in the file view have been expanded.

**Expected behavior**
Icons should remain visible once set. not be replaced by their text code.

**Screenshots**
<img width="283" alt="image" src="https://user-images.githubusercontent.com/40892865/162194862-be73ac49-54cd-4ee8-951c-16e4e93633d5.png">
here you can see how some of the folder icons are gone, but not all

<img width="196" alt="Screen Shot 2022-04-07 at 14 07 33" src="https://user-images.githubusercontent.com/40892865/162195147-351d4a29-d5ad-4578-9a27-84bc1d07ef19.png">
also happens with the Remix icons

**Additional context**
- This also happens on mobile and my ipad.
- I am saving the icons in .obsidian/icons on each device. to prevent issues with obsidian sync.

**Comment by ChristinWhite** [comment]
It's happening to me too, the icons are still in placed, I just have to relink them.

**Comment by FlorianWoelki** [comment]
Thank you for reporting. Unfortunately, I cannot reproduce this at the moment. Because you can easily relink them, the icons (SVG files) are still in their directories. Therefore, it should be some kind of issue with the loading of the icons or the syncing. I will inspect this issue.

**Comment by ChristinWhite** [comment]
Sorry for the slow follow up, I'm going to try to do some testing and see if I can better narrow down the issue Tuesday or Wednesday (my weekend) and get you some better information.

**Comment by takeo** [comment]
I've run into this a lot as well. It seems to be something to do with syncing between two machines using Obsidian Sync. I've sometimes deleted and reinstalled the icon packs, which helps, but it also usually requires a lot of re-changing the icons on one of the machines. Also, I usually stick to all line style, but they sometimes disappear so I switch to solid, then back again when it happens in the future.

Let me know if you need any Obsidian Sync logs or anything to see if there's some sort of conflict happening. Also, it's tons of small SVGs so not sure if that can get weird with sync.

**Comment by ChristinWhite** [comment]
Okay, so here's some additional details that may help, it looks to me like this may be an issue with specific icon packs that you didn’t include as options to download in the plugin. 

- **Syncing**: Yes, I’m using Obsidian Sync. I have it syncing everything, including images, other file types, main settings, active community plugin list and installed community plugins
- **Randomly**: It seems to happen only after loading/reloading a vault, I don't see it happening 'randomly' like I first thought though I wouldn't call it definitive.
- **Icons**:
	- It's the same icons every time on macOS. It's the same every time on iOS but it's not entirely the same on both platforms. Every icon that disappears on macOS also vanishes on iOS but additional icons also vanish.
	- It seems to be consistent on which icon packs the disappearing icons are from, mainly [Octicons](https://github.com/primer/Octicons) and [Zondicons](https://www.zondicons.com/). On iOS it seems like all Font Awesome icons are fine, some but not all Remix icons load.
- **Plugin**: `data.json` is not changing on launch, nor after relinking them outside of `"recentlyUsedIcons":[]` which I tend to have shift as I relink them.
- **Loading**:
	- I do not get the ‘Seems like you do not have an icon pack installed (*icon-prefix-name*).
	- I tried removing all icon packs except for Octicons just to see if it might be trying to go through too many icons but the Octicon icons are still unlinked.
- **Version**: I think this started happening on version 1.1.0 when you revamped the icons available but that was also probably what drove me to add a bunch of other icon packs to my setup so it may be correlation, not causation.
- **iOS note**: In the plugin options on iOS only the Octicons and remix-icons folders are shown ([screenshot](https://user-images.githubusercontent.com/1746405/163286729-af205f7d-4653-4d86-9eaf-3a7bbb925477.png)), not the other 8 icon packs, despite the files showing up in iMazing and not throwing the missing error. This could be an unrelated bug.
- **Icon Differences**: I did compare a Font Awesome icon that always works and a octicon icon that always vanishes.
	- Similar file sizes.
	- Octicon used a 16pixel document size while Font Awesome did 512px.
	- They both have a single path.
	- Same `xmlns`, both include a `viewBox`.
	- Octicons includes `width` and `height`, Font Awesome omits it.
	- Font Awesome omits any fill, Octicons uses `fill-rule="evenodd`.

If you’d like a zip of all of my `.obsidian/icons` and my `obsidian-icon-folder` let me know where to message you and I’ll send you a dropbox link, I don’t want to break any licenses uploading them publicly. All are freely available (except a couple of my own icons).

Let me know if there’s anything else you’d like me to provide!

**Comment by dartungar** [comment]
This happens for me, too. Usually I don't mind, but with icon pack of 2000+ icons it brings sync to a halt because it tries to re-load all the icons for mobile.

**Comment by FlorianWoelki** [comment]
> Okay, so here's some additional details that may help, it looks to me like this may be an issue with specific icon packs that you didn’t include as options to download in the plugin, at least in my case.
> 
> 
> 
> - **Syncing**: Yes, I’m using Obsidian Sync. I have it syncing everything, including images, other file types, main settings, active community plugin list and installed community plugins
> 
> - **Randomly**: It seems to happen only after loading/reloading a vault, I don't see it happening 'randomly' like I first thought though I wouldn't call it definitive.
> 
> - **Icons**:
> 
> 	- It's the same icons every time on macOS. It's the same every time on iOS but it's not entirely the same on both platforms. Every icon that disappears on macOS also vanishes on iOS but additional icons also vanish.
> 
> 	- It seems to be consistent on which icon packs the disappearing icons are from, mainly [Octicons](https://github.com/primer/Octicons) and [Zondicons](https://www.zondicons.com/). On iOS it seems like all Font Awesome icons are fine, some but not all Remix icons load.
> 
> - **Plugin**: `data.json` is not changing on launch, nor after relinking them outside of `"recentlyUsedIcons":[]` which I tend to have shift as I relink them.
> 
> - **Loading**:
> 
> 	- I do not get the ‘Seems like you do not have an icon pack installed (*icon-prefix-name*).
> 
> 	- I tried removing all icon packs except for Octicons just to see if it might be trying to go through too many icons but the Octicon icons are still unlinked.
> 
> - **Version**: I think this started happening on version 1.1.0 when you revamped the icons available but that was also probably what drove me to add a bunch of other icon packs to my setup so it may be correlation, not causation.
> 
> - **iOS note**: In the plugin options on iOS only the Octicons and remix-icons folders are shown ([screenshot](https://user-images.githubusercontent.com/1746405/163286729-af205f7d-4653-4d86-9eaf-3a7bbb925477.png)), not the other 8 icon packs, despite the files showing up in iMazing and not throwing the missing error. This could be an unrelated bug.
> 
> - **Icon Differences**: I did compare a [Font Awesome icon](https://fontawesome.com/v5/icons/app-store-ios?s=brands) that always works and an [Octicon icon](https://github.com/primer/octicons/blob/main/icons/rocket-16.svg) that always vanishes, none of it seems significant to me.
> 
> 	- Similar file sizes.
> 
> 	- Octicon used a 16px document size while Font Awesome did 512px.
> 
> 	- They both have a single path.
> 
> 	- Same `xmlns`, both include a `viewBox`.
> 
> 	- Octicons includes `width` and `height`, Font Awesome omits it.
> 
> 	- Octicons uses `fill-rule="evenodd`, Font Awesome omits it as well.
> 
> 
> 
> If you’d like a zip of all of my `.obsidian/icons` and my `obsidian-icon-folder` let me know where to message you and I’ll send you a dropbox link, I don’t want to break any licenses uploading them publicly. All are freely available (except a couple of my own icons).
> 
> 
> 
> Let me know if there’s anything else you’d like me to provide!

Thank you for the detailed explanation! Can you provide me a minimal reproduction vault and send it to the discord profile FloWy#3979

**Comment by ChristinWhite** [comment]
Just got it together and sent you a friend request so I can send it over.

**Comment by FlorianWoelki** [comment]
So I added a new button called 'Fix Icon Pack' in the settings. Some icons disappear because they are not normalized like all the icons in Octions and Zondicons. I will release it soon.

**Comment by ChristinWhite** [comment]
Interesting, thanks for investigating! Do you mind if I ask what normalizing it means in this context, was it the included width and height?

I'm working on some of my own icons for Obsidian and want to make sure they'll be compatible without needing changes.

**Comment by FlorianWoelki** [comment]
> Interesting, thanks for investigating! Do you mind if I ask what normalizing it means in this context, was it the included width and height?
> 
> I'm working on some of my own icons for Obsidian and want to make sure they'll be compatible without needing changes.

The issue that Octions and Zondicons did not work with was that the file names were not "recognized" as icons. However, the "fix" button does now try to rename the files such that they always fit.

They should be compatible, no worries.

**Comment by FlorianWoelki** [comment]
With the newest release 1.1.6 the "fix icon pack" button is now included. Because the issue depended on syncing and normalizing issues, I will close this one for now. Please reopen a new issue whenever there is a new problem. Thank you all for reporting and thank you especially @ChristinWhite for the detailed report!
